dhis2-mobile-program-rules/src/main/java/org/dhis2/mobileProgramRules/RuleEngineExtensions.kt

```diff
@@ -111,7 +111,7 @@ fun ProgramRuleAction.toRuleEngineObject(): RuleAction {
                 values = mutableMapOf(
                     Pair("field", field),
                 ).also { map ->
-                    content()?.let { map["content"] = it }
+                    (displayContent() ?: content())?.let { map["content"] = it }
```

Comment:
This property, is it evaluated for all the action types? If so, we could evaluate only once at the beginning of the method (out of the `when`, next to `field`).